Underlinking in libOgreBites.so on Linux X11

#### System Information
- Ogre Version: 1.11.0
- Operating System / Platform: Mageia 6 x86_64 / Linux
- RenderSystem: n/a (build issue)

#### Detailled description
I am building Ogre 1.11.0 with the following link flags to prevent [underlinking](https://wiki.mageia.org/en/Underlinking_issues_in_packaging): `-Wl,--as-needed -Wl,--no-undefined` (as per policy when packaging for the Mageia distro).

It raises a linking issue in libOgreBites.so on Linux, which uses X11 APIs without linking them:
```
[ 78%] Linking CXX shared library ../../lib/libOgreBites.so
cd /home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/build/Components/Bites && /usr/bin/cmake -E cmake_link_script CMakeFiles/OgreBites.dir/link.txt --verbose=1
/bin/c++ -fPIC -Wall -Winit-self -Wcast-qual -Wwrite-strings -Wextra -Wundef -Wmissing-declarations -Wno-unused-parameter -Wshadow -Wno-missing-field-initializers -Wno-long-long -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -msse -DNDEBUG -Wl,--as-needed -Wl,--no-undefined -Wl,-z,relro -Wl,-O1 -Wl,--build-id -Wl,--enable-new-dtags -shared -Wl,-soname,libOgreBites.so.1.11.0 -o ../../lib/libOgreBites.so.1.11.0 CMakeFiles/OgreBites.dir/src/OgreAdvancedRenderControls.cpp.o CMakeFiles/OgreBites.dir/src/OgreApplicationContext.cpp.o CMakeFiles/OgreBites.dir/src/OgreBitesConfigDialog.cpp.o CMakeFiles/OgreBites.dir/src/OgreCameraMan.cpp.o CMakeFiles/OgreBites.dir/src/OgreSGTechniqueResolverListener.cpp.o CMakeFiles/OgreBites.dir/src/OgreStaticPluginLoader.cpp.o CMakeFiles/OgreBites.dir/src/OgreTrays.cpp.o CMakeFiles/OgreBites.dir/src/OgreWindowEventUtilities.cpp.o CMakeFiles/OgreBites.dir/src/OgreGLXConfigDialog.cpp.o ../../lib/libOgreOverlay.so.1.11.0 /usr/lib64/libXt.so /usr/lib64/libXaw.so ../../lib/libOgreRTShaderSystem.so.1.11.0 /usr/lib64/libSDL2.so -lpthread ../../lib/libOgreMain.so.1.11.0 -lpthread 
CMakeFiles/OgreBites.dir/src/OgreWindowEventUtilities.cpp.o: In function `OgreBites::WindowEventUtilities::messagePump()':
/home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/Components/Bites/src/OgreWindowEventUtilities.cpp:82: undefined reference to `XCheckWindowEvent'
/home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/Components/Bites/src/OgreWindowEventUtilities.cpp:88: undefined reference to `XCheckTypedWindowEvent'
CMakeFiles/OgreBites.dir/src/OgreGLXConfigDialog.cpp.o: In function `OgreBites::GLXConfigurator::~GLXConfigurator()':
/home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/Components/Bites/src/OgreGLXConfigDialog.cpp:203: undefined reference to `XCloseDisplay'
/home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/Components/Bites/src/OgreGLXConfigDialog.cpp:197: undefined reference to `XFreePixmap'
CMakeFiles/OgreBites.dir/src/OgreGLXConfigDialog.cpp.o: In function `OgreBites::GLXConfigurator::CreateBackdrop(unsigned long, int)':
/home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/Components/Bites/src/OgreGLXConfigDialog.cpp:358: undefined reference to `XCreateGC'
/home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/Components/Bites/src/OgreGLXConfigDialog.cpp:364: undefined reference to `XCreateImage'
/home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/Components/Bites/src/OgreGLXConfigDialog.cpp:373: undefined reference to `XCreatePixmap'
/home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/Components/Bites/src/OgreGLXConfigDialog.cpp:377: undefined reference to `XPutImage'
/home/akien/Mageia/Checkout/ogre/BUILD/ogre-1.11.0/Components/Bites/src/OgreGLXConfigDialog.cpp:381: undefined reference to `XFreeGC'
collect2: error: ld returned 1 exit status
Components/Bites/CMakeFiles/OgreBites.dir/build.make:311: recipe for target 'lib/libOgreBites.so.1.11.0' failed
```

Adding `-lX11` to the linker should fix it.

#### Ogre.log
n/a

--- a/Components/Bites/CMakeLists.txt
+++ b/Components/Bites/CMakeLists.txt
@@ -103,7 +103,7 @@ elseif(APPLE)
   set(DEPENDENCIES ${DEPENDENCIES} "-framework Cocoa")
 elseif(UNIX)
   list(APPEND SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/src/OgreGLXConfigDialog.cpp")
-  list(APPEND DEPENDENCIES ${X11_Xt_LIB} ${XAW_LIBRARY})
+  list(APPEND DEPENDENCIES ${X11_X11_LIB} ${X11_Xt_LIB} ${XAW_LIBRARY})
   install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/misc/GLX_backdrop.png" DESTINATION "share/OGRE/")
 endif()
 

